src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -94,11 +107,14 @@ public sealed class ContainerStepHost : RunnerService, IContainerStepHost
         public event EventHandler<ProcessDataReceivedEventArgs> OutputDataReceived;
         public event EventHandler<ProcessDataReceivedEventArgs> ErrorDataReceived;
 
-        public string ResolvePathForStepHost(string path)
+        public string ResolvePathForStepHost(IExecutionContext executionContext, string path)
         {
             // make sure container exist.
             ArgUtil.NotNull(Container, nameof(Container));
-            ArgUtil.NotNullOrEmpty(Container.ContainerId, nameof(Container.ContainerId));
+            if (!FeatureFlagManager.IsContainerHooksEnabled(executionContext))
+            {
+                ArgUtil.NotNullOrEmpty(Container.ContainerId, nameof(Container.ContainerId));
```

Comment:
Should we just consider moving this check? Maybe a todo in the future

Reply:
Or just pass in a bool rather then passing in the full context

Reply:
What do you think of passing `executionContext.Globals.Variables`? It gives us the access to server sent FFs.